### ISSUE TITLE
One Alert Pager Duty + Slack

### ISSUE THREAD

**Issue by QaysarA** [department-of-veterans-affairs/notification-api#223]
# Value Statements
**As** a VANotify developer
**I want to** be alerted when resource cost is high
**So that** we can make sure we are not expending unnecessary costs.

**As** a VANotify developer
**I want to** be alerted whens systems are down or not functioning as expected
**So that** someone on duty can fix it.

# Acceptance Criteria
**GIVEN** an alert notifies the team about a concern
**WHEN** the developer receives the notification via PagerDuty
**THEN** they are informed enough to act on the notification.

**GIVEN** an alert notifies the team about a concern
**THEN** an alert is sent via Slack.

**GIVEN** an alert needs to be changed
**THEN** a developer can update the code to update the dashboards

# Checklist:
- [ ] DataDog Set Up
    - [ ] Create a dashboard of metrics, [using terraform](https://docs.datadoghq.com/api/v1/dashboards/)
    - [ ] Configure alerts (see below)
- [ ] PagerDuty
    - [ ] Get access from VA - contact DOTS team
    - [ ] Set up team and schedule
    - [ ] All team members add their notification preference (email/text)
    - [ ] Alerting is [integrated with PagerDuty](https://docs.datadoghq.com/api/v1/pagerduty-integration/)
- [ ] Slack - to private #notifications channel
    - [ ] [DataDog Slack Integration](https://docs.datadoghq.com/api/v1/slack-integration/)

# Alerts to set up
- SQS: QAYSAR + Marisa - 1. Would be the first (includes PagerDuty, Slack) Estimate (4), all 2. other alerts in a separate card <add depen on previous card (3)
   
- App Metrics:
    - [ ] Measure weather the heath check is returning success or failure. on failure it should give an alert

# Assumptions
- ~For now, we will set up alerts with CloudWatch and move to DataDog later.~ Since we want to move to DataDog eventually and have the alerts all in one place, we should set them up in DataDog now instead of putting work into CloudWatch first and migrating.
- Determining the various thresholds/numbers may require setting up the metrics and measuring what is typical of our system or further researching measures like cost.

# Additional Info/Resources
-  Research from the spike #151 in [Google doc](https://docs.google.com/document/d/1gwaBA9jC__IYMLcdpQjVPw3UjsKy1GVOjhtQa2nSEY8/edit), explaining the reasoning and how to set up each alert.
- Helpful [PagerDuty setup video](https://www.youtube.com/watch?v=eGUjzjkOwSw)
- [DataDog monitor](https://registry.terraform.io/providers/DataDog/datadog/latest/docs/resources/monitor) terraform
- Refer to ADR 6 in `vanotify-team` repo

# Out of Scope
-

# Open Questions
-  Which alerts warrant a notification via PagerDuty versus a Slack message? Should a slack message always be sent?
-  What is our on-call procedure?
- If metrics are set up in DataDog through pushing code, how do environments factor into this? (if we haven't pushed to prod, does the dashboard reflect the new metrics added for prod?)
- Do we only want to see metrics/get alerts for production?

**Comment by philherbert** [comment]
Blocked while we're waiting on getting access to pagerduty

**Comment by miabecker** [comment]
Seems like there is a lag in datadog now so need to consider cloud watch @miabecker
